src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -233,6 +233,65 @@ public void cleanup(Long id, boolean deleteRepositories) throws IOException {
         }
     }
 
+    @Transactional(readOnly = true)
+    public java.io.File exportParticipations(Long id, List<String> studentIds) {
+        Exercise exercise = findOneLoadParticipations(id);
+        List<Path> zippedRepoFiles = new ArrayList<>();
+        Path zipFilePath = null;
+        if (Optional.ofNullable(exercise).isPresent() && exercise instanceof ProgrammingExercise) {
+            exercise.getParticipations().forEach(participation -> {
+                try {
+                    if (participation.getRepositoryUrl() != null && studentIds.contains(participation.getStudent().getLogin())) {
+                        boolean repoAlreadyExists = gitService.get().repositoryAlreadyExists(participation.getRepositoryUrlAsUrl());
+
+                        Repository repo = gitService.get().getOrCheckoutRepository(participation);
+                        log.info("Create temporary zip file for repository " + repo.getLocalPath().toString());
+                        Path zippedRepoFile = gitService.get().zipRepository(repo);
+                        zippedRepoFiles.add(zippedRepoFile);
+                        boolean allowInlineEditor = ((ProgrammingExercise) exercise).isAllowOnlineEditor() != null && ((ProgrammingExercise) exercise).isAllowOnlineEditor();
+                        if(!allowInlineEditor){ //if onlineeditor is not allowed we are free to delete
+                            log.info("Delete temporary repoistory "+ repo.getLocalPath().toString());
```

Comment:
please change all logs to debug level, so that they don't pollute the logs on the production server